src/borg/archiver.py

```diff
@@ -652,126 +709,82 @@ def _process(self, *, path, parent_fd=None, name=None,
                 with OsOpen(path=path, parent_fd=parent_fd, name=name, flags=flags_dir,
                             noatime=True, op='dir_open') as child_fd:
                     # child_fd is None for directories on windows, in that case a race condition check is not possible.
                     if child_fd is not None:
                         with backup_io('fstat'):
                             st = stat_update_check(st, os.fstat(child_fd))
-                    if recurse:
+
+                    if right_dev:
                         tag_names = dir_is_tagged(path, exclude_caches, exclude_if_present)
                         if tag_names:
                             # if we are already recursing in an excluded dir, we do not need to do anything else than
                             # returning (we do not need to archive or recurse into tagged directories), see #3991:
-                            if not recurse_excluded_dir:
-                                if keep_exclude_tags and not dry_run:
-                                    fso.process_dir(path=path, fd=child_fd, st=st)
-                                    for tag_name in tag_names:
-                                        tag_path = os.path.join(path, tag_name)
-                                        self._process(path=tag_path, parent_fd=child_fd, name=tag_name,
-                                                      fso=fso, cache=cache, matcher=matcher,
-                                                      exclude_caches=exclude_caches, exclude_if_present=exclude_if_present,
-                                                      keep_exclude_tags=keep_exclude_tags, skip_inodes=skip_inodes,
-                                                      restrict_dev=restrict_dev, read_special=read_special, dry_run=dry_run)
-                                self.print_file_status('x', path)
+                            if recurse_excluded_dir:
+                                return
+                            if keep_exclude_tags:
+                                status = self._process_any(path=path, parent_fd=parent_fd, name=name, st=st, fso=fso,
+                                                      cache=cache, read_special=read_special, dry_run=dry_run)
+                                for tag_name in tag_names:
+                                    tag_path = os.path.join(path, tag_name)
+                                    self._rec_walk(
+                                            path=tag_path, parent_fd=child_fd, name=tag_name, fso=fso, cache=cache,
+                                            matcher=matcher, exclude_caches=exclude_caches, exclude_if_present=exclude_if_present,
+                                            keep_exclude_tags=keep_exclude_tags, skip_inodes=skip_inodes,
+                                            restrict_dev=restrict_dev, read_special=read_special, dry_run=dry_run)
+                            self.print_file_status('x', path)
```

Comment:
did you just do 745/746 to get out more quickly?


if so and there is no functional change, please keep the old way as it is easier to review for now (smaller diff).

Reply:
I think i did it mainly to avoid deep nesting. It's definitely style, i'll revert that.